packages/legend-application-studio/src/stores/editor/sidebar-state/BulkServiceRegistrationState.ts

```diff
@@ -33,12 +40,39 @@ import type { ServiceRegistrationEnvironmentConfig } from '../../../application/
 
 export const LATEST_PROJECT_REVISION = 'Latest Project Revision';
 
+export enum REGISTRATION_RESULT {
+  DID_NOT_RUN = 'DID_NOT_RUN',
+  FAILED = 'FAILED',
+  SUCCESS = 'SUCCESS',
+  IN_PROGRESS = 'IN_PROGRESS',
+}
+
+export const getServiceRegistrationResult = (
+  isServiceRegistering: boolean,
+  result: ServiceRegistrationResult | undefined,
+): REGISTRATION_RESULT => {
+  if (isServiceRegistering) {
+    return REGISTRATION_RESULT.IN_PROGRESS;
+  } else if (result instanceof ServiceRegistrationSuccess) {
+    return REGISTRATION_RESULT.SUCCESS;
+  } else if (result instanceof ServiceRegistrationFail) {
+    return REGISTRATION_RESULT.FAILED;
+  }
+  return REGISTRATION_RESULT.DID_NOT_RUN;
+};
+
 export class BulkServiceRegistrationState extends ServiceConfigState {
```

Comment:
why odes this extends ServiceConfigState. If i understand correctly this just holds the service state for registration. either if its selected or the result.

Reply:
I would have to copy less code into BulkServiceRegistrationState because I reference ServiceConfigState.options and serviceEnv

Reply:
but you have access to the config 
through `this.globalBulkServiceRegistrationState.config`. if you need information about it
I dont thinkt he inhertiance here makes sense

Reply:
I understand, I got rid of the extends ServiceConfigState